feat(user-profile-summary): update styles to match design

arathjz opened the pull request:
**What:**
- Update styles on user summary to match the design

**Why:**
To address: https://app.asana.com/0/1159164196409129/1186341795801831/f subtask 4

**How:**
- Overwrite summary.hbs to customize the order of summary stats
- Overwrite default styles from discourse to use our own design

**Notes**
- Top categories and top badges were not included in the Figma design, so, I did remove them from the UI. If they need to be displayed let me know to add them.

![image](https://user-images.githubusercontent.com/20747535/89594343-2718b700-d817-11ea-8d4d-806dc0cf5ec5.png)

Media:
https://www.loom.com/share/434732bf989a4218b7a2008840ff8239
![image](https://user-images.githubusercontent.com/20747535/89594456-76f77e00-d817-11ea-8109-8517de2b9d42.png)
![image](https://user-images.githubusercontent.com/20747535/89594441-69da8f00-d817-11ea-923f-daaf7c8e0b4c.png)

duranmla commented on scss/templates/user.scss:
```diff
@@ -198,3 +148,188 @@
+    }
+  }
+
+  @include media-breakpoint-down(sm) {
```
Remember to enforce yourself to work mobile first. `media-breakpoint-up`

duranmla commented on scss/templates/user.scss:
```diff
@@ -198,3 +148,188 @@
+    .number,
+    .value {
+      font-size: 2.25rem;
+      font-family: "Roboto", $font-family-sans-serif;
```
You really need this? I wonder:

1. Do we have a different font family for a set of things within the admin? If so maybe is a mistake
2. Can we use variables that we already define with the font family?
3. Those this text can benefit from something like `@extend .dc-text-monospace` _(take a look to `type.scss`)_

arathjz replied:
I took this from the design

![image](https://user-images.githubusercontent.com/20747535/89696583-2eef5e80-d8de-11ea-8e98-09d86ec31c21.png)

I could use monospace 💯

arathjz replied:
Monospace looks very different from the designs

![image](https://user-images.githubusercontent.com/20747535/89696621-55ad9500-d8de-11ea-9e0e-b4d38b07cfa0.png)

I think for this case we could still be using `Libre Franklin`

arathjz replied:
This is how it looks with Libre Franklin 

![image](https://user-images.githubusercontent.com/20747535/89696658-855c9d00-d8de-11ea-8d62-cccb712fe9b0.png)

orlando replied:
I guess we can use the Libre Frankling font here, looks good to me.

duranmla commented on scss/templates/user.scss:
```diff
@@ -198,3 +148,188 @@
+      border-bottom: 0.0625rem solid $gray-ligth;
+      border-left: none;
+
+      &:not(:last-child) {
```
Why you need the ":not(:last-child)" more specificity? Cause with `&:last-child {` you are already set

arathjz replied:
You're so right, good catch

duranmla commented on scss/templates/user.scss:
```diff
@@ -198,3 +148,188 @@
+    width: 100%;
+    margin-bottom: 0;
+
+    > div > ul > li,
```
I will rather add more specificity like:

```
.top-sub-section,
.summary-user-list {
    &.summary-user-list li,
    &.top-sub-section li {
        //...
    }   
}
```

Still a bit odd but we are not that tight to DOM structure. _(same for other rules with similar approach)_

duranmla commented on scss/templates/user.scss:
```diff
@@ -198,3 +148,188 @@
+        justify-content: space-between;
+        align-items: center;
+        padding: 0;
+        width: 80%;
```
_Suspicious..._ do you need the `80%` here?

arathjz replied:
This is for using the maximum space available on a stat item, removing this rule we would have this

![image](https://user-images.githubusercontent.com/20747535/89696412-66a9d680-d8dd-11ea-8934-0849ee58c616.png)

Would I should add a comment for this line?

orlando replied:
Mmm maybe we should use paddings instead?

arathjz replied:
@orlando not sure, by default, discourse has this style rule

![image](https://user -images.githubusercontent.com/20747535/89716016-707c1a00-d96f-11ea-84e8-b34be37ec9ae.png)

I only added a 10% so it looked a little bit wider

Let me know what you guys think

duranmla replied:
We can afterwards revisit together. The idea here would be to have a flex container and define a width within the smaller component so the other items will _"naturally"_ fit within the whole container. But we can check it out together just to avoid the need of doing this in the future.

_For now we can move on and not stop the pull request due to this_

arathjz commented:
@duranmla thanks for your review! 💯 I've addressed some of your comments.

Regarding the template, my first approach was to only add changes on css, however, some elements were really nested and the element's structure was not really flexible to only add css changes. I decided to overwrite the template since I thought that would be a better way to maintain. 

Let me know what you think and we can discuss this on a further meeting if needed 🚀
